fabric-resource-conditions-api-v1/src/main/java/net/fabricmc/fabric/api/resource/conditions/v1/DefaultResourceConditions.java

```diff
@@ -183,6 +240,19 @@ static void init() {
 		ResourceConditions.register(ITEM_TAGS_POPULATED, object -> ResourceConditionsImpl.tagsPopulatedMatch(object, RegistryKeys.ITEM));
 		ResourceConditions.register(TAGS_POPULATED, ResourceConditionsImpl::tagsPopulatedMatch);
 		ResourceConditions.register(FEATURES_ENABLED, ResourceConditionsImpl::featuresEnabledMatch);
+		ResourceConditions.register(REGISTRY_CONTAINS, ResourceConditionsImpl::registryContainsMatch);
+	}
+
+	// Slightly gross - the empty outputType vararg is used to capture the correct type for B[]
+	@SafeVarargs
+	private static <A, B> B[] transform(A[] input, Function<A, B> mapper, B... outputType) {
+		B[] output = Arrays.copyOf(outputType, input.length);
+
+		for (int i = 0; i < input.length; i++) {
+			output[i] = mapper.apply(input[i]);
+		}
+
+		return output;
 	}
```

Comment:
I know this is ugly, but it makes the vararg manipulation a lot shorter, which makes the behavior of the various factory methods much easier to follow for someone reading through them the first time. Worth it imo. :wink: